lib/payloadAction.ts

```diff
@@ -77,6 +77,15 @@ export const  handlePayloadActions = async (app: IApp, read: IRead,  modify: IMo
                     }
                 }
             }
+        } else {
+            // widechat specific
+            const textMessages: Array<string | IDialogflowQuickReplies | IDialogflowPayload |  IDialogflowImageCard> = [];
+            textMessages.push(message);
+            const messagesToProcess: IDialogflowMessage = {
+                messages: textMessages,
```

Comment:
Calling `createDialogflowMessage` in `handlePayloadActions` function is not a good idea. Only job of `handlePayloadActions` should be to handle payload action.

If required we should create, separate fn to `handleResponse`, which will then call `handlePayloadActions` and `createDialogFlowMessage` functions based on response type. (If action call  `handlePayloadActions` else `createDialogFlowMessage`)